apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/util/RoleUtils.java

```diff
@@ -107,7 +107,7 @@ public static String buildNamespaceTargetId(String appId, String namespaceName)
   }
 
   public static String buildNamespaceTargetId(String appId, String namespaceName, String env) {
-    return STRING_JOINER.join(appId, namespaceName, env);
+    return STRING_JOINER.join(appId, env, namespaceName);
```

Comment:
Unfortunately, we can't make this change because it would break backward compatibility.